### PR TITLE
fix: remove unqualified did from out of band record

### PR DESCRIPTION
**Pull request by TimoGlastra**
And various other fixes from #764

### CODE REVIEW

File: packages/core/src/modules/connections/services/ConnectionService.ts [hunk 1]
```diff
@@ -39,6 +40,8 @@ import {
   Ed25119Sig2018,
   EmbeddedAuthentication,
```
Comment (blu3beri) [inline packages/core/src/modules/connections/services/ConnectionService.ts:41]: unused import.

File: packages/core/src/modules/connections/services/ConnectionService.ts [hunk 2]
```diff
@@ -615,24 +602,28 @@ export class ConnectionService {
...
     autoAcceptConnection?: boolean
     multiUseInvitation: boolean
     tags?: CustomConnectionTags
+    threadId?: string
     imageUrl?: string
     protocol?: HandshakeProtocol
     outOfBandId?: string
     invitationDid?: string
   }): Promise<ConnectionRecord> {
     const connectionRecord = new ConnectionRecord({
```
Comment (blu3beri) [inline packages/core/src/modules/connections/services/ConnectionService.ts:618]: why can't we pass `options` to this class? this seems a tad bit too verbose :)

File: packages/core/src/modules/oob/OutOfBandModule.ts [hunk 3]
```diff
@@ -216,28 +221,15 @@ export class OutOfBandModule {
    *
    * Agent role: sender (inviter)
    *
-   * @param config configuration of how out-of-band invitation should be created
+   * @param config configuration of how connection invitation should be created
```
Comment (blu3beri) [inline packages/core/src/modules/oob/OutOfBandModule.ts:224]:
```suggestion
   * @param config configuration of how a connection invitation should be created
```

**Review by blu3beri — APPROVED** [review]
Again, just some nits about the code nothing blocking or major.

File: packages/core/src/agent/MessageSender.ts [hunk 4]
```diff
@@ -212,7 +212,14 @@ export class MessageSender {
       outOfBand
     )
 
-    const ourDidDocument = await this.resolveDidDocument(connection.did)
+    if (!connection.did) {
+      this.logger.error(`Unable to send message using connection '${connection.id}' that doesn't have a did`)
+      throw new AriesFrameworkError(
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply packages/core/src/agent/MessageSender.ts:217]: Issue created

File: packages/core/src/agent/MessageSender.ts [hunk 5]
```diff
@@ -358,7 +365,7 @@ export class MessageSender {
         // Resolve dids to DIDDocs to retrieve routingKeys
         const routingKeys = []
         for (const routingKey of didCommService.routingKeys ?? []) {
-          const routingDidDocument = await this.resolveDidDocument(routingKey)
+          const routingDidDocument = await this.didResolverService.resolveDidDocument(routingKey)
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply packages/core/src/agent/MessageSender.ts:368]: Do you mean using the module here? It's already possible to resolve dids from the public api `agent.did.resolve()`.

**Pull request merged**